app/src/main/kotlin/edu/stanford/bdh/engagehf/bluetooth/BluetoothViewModel.kt

```diff
@@ -54,6 +56,7 @@ class BluetoothViewModel @Inject internal constructor(
         observeBleService()
         observeRecords()
         observeMessages()
+        viewModelScope.launch { deviceRegistrationService.registerDevice() }
```

Comment:
I would not mark the `registerDevice` as suspend and launch  `ioScope` internally instead. The reason is that we want that operation to finish and not bound it to the lifecycle of this view model.

QQ: Is this manual call necessary? Can't we just rely on `onNewToken` in the service? From the docs: `Called when a new token for the default Firebase project is generated.
This is invoked after app install when a token is first generated, and again if the token changes.` The `DeviceRegistrationService` can then also be marked internal. We can then also update it to `registerDevice(token: String)` and call it with the new token, `MessageTokenService` can also be removed then as not needed

Reply:
I called it on every startup because the RegisterDevice method also updates other things besides the token, such as the language and time zone. But I guess these are things that change so rarely that it is negligible. i have adapted it according to the suggestion